src/IndexMaps/realindexmap.jl

```diff
@@ -75,12 +76,98 @@ function calculate_ind_values(imap::RealIndexMap, xs::Vector, dims::Vector{Int})
   return ind_to_ind_value_map
 end
 
-function grid_points(imap::RealIndexMap, N::Int, d::Int)
+""" 
+    grid_points(imap, N, d, span; exact_grid=true, enforced=[])
+
+  Gives `N` grid points from a given dimension of `imap` within a specified range.
+
+  # Arguments
+  - `imap::RealIndexMap`: An IndexMap specifying the structure of the TN being used
+  - `N::Int`: The number of grid points requested.
+  - `d::Int` The index of the dimension of `imap` requested
+  - `span::AbstractVector{<:Number}`: A two element number vector [a,b] with 0≤a<b≤1. The right endpoint of this span is not included as a gridpoint in the output.
+  - `exact_grid::Bool`: Flag specifying whether the function should give exact grid points 
+  (note: using `exact_grid=true` may cause less than `N` grid points to be returned)
+  - `enforced::AbstractVector{<:Number}`: A list of points that we want to enforce to show up in the grid.
+"""
+function grid_points(
+  imap::RealIndexMap,
+  N::Int,
+  d::Int;
+  span::AbstractVector{<:Number}=[0, 1],
+  exact_grid::Bool=true,
+  enforced=[],
+)
+  if length(span) != 2 || span[1] >= span[2] || span[1] < 0 || span[2] > 1
+    throw(
+      "expected a two-element vector [a,b] with 0≤a<b≤1 as input span, instead found $span"
+    )
+  end
+  dims = dim.(dimension_inds(imap, d))
+  @assert all(y -> y == first(dims), dims)
+  base = first(dims)
+  L = length(dimension_inds(imap, d))
+
+  # generate grid_points within the span (exclusive of right endpoint)
+  grid_points = collect(range(span[1], span[2] - (span[2] - span[1]) / N; length=N))
+  if exact_grid
+    for (i, point) in enumerate(grid_points)
+      grid_points[i] = round_to_nearest_exact_point(point, L)
+    end
+    # define lambda function
+    # round_near = point -> round_to_nearest_exact_point(point, L)
+    # grid_points = map(round_near, grid_points)
+    grid_points = unique(grid_points)
+  end
+
+  # add enforced points
+  if !isempty(enforced)
+    grid_points = sort(unique(vcat(grid_points, enforced_points)))
+  end
+
+  return grid_points
+end
+
+function round_to_nearest_exact_point(point::Number, L::Int)
+  if point < 0 || point >= 1
```

Comment:
I think we can remove this check. I prefer to not have too many error catching exceptions, an error will get thrown eventually anyway